prog.go

```diff
@@ -155,14 +155,20 @@ func (p *BPFProg) SetAttachTarget(attachProgFD int, attachFuncName string) error
 	return nil
 }
 
-// TODO: fix API to return error
-func (p *BPFProg) SetProgramType(progType BPFProgType) {
-	C.bpf_program__set_type(p.prog, C.enum_bpf_prog_type(int(progType)))
+func (p *BPFProg) SetProgramType(progType BPFProgType) error {
+	retC := C.bpf_program__set_type(p.prog, C.enum_bpf_prog_type(int(progType)))
+	if retC < 0 {
+		return fmt.Errorf("failed to set prog_type %s for program %s", progType.String(), p.Name())
+	}
+	return nil
```

Comment:
nit: place an empty line before the default return as standardised in the rest of the file.